core/thread/id.go

```diff
@@ -276,6 +290,13 @@ func (i ID) MarshalText() ([]byte, error) {
 	return []byte(i.String()), nil
 }
 
+func (i ID) MarshalJSON() ([]byte, error) {
+	data := map[string]string{
+		"str": i.String(),
+	}
+	return json.Marshal(data)
```

Comment:
Other part of this solution is you actually have to return json data from `MarshalJSON` (i.e. a `map` in go). If you return bytes encoding some string value, the schema validation fails because the marshaled json you are validating includes a `string` value where the schema says the value must be an `object`. Returning bytes encoding a map here makes sure that is the case.